iot-hub/Samples/service/DigitalTwinClientSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -59,77 +65,83 @@ private async Task UpdateDigitalTwinComponentPropertyAsync()
+                    _logger.LogDebug($"The property {targetTemperaturePropertyName} under component {Thermostat1Component} `" +
+                        $"was never set on the {_digitalTwinId} digital twin.");
 
                     // The property path to be added should be prepended with a '/'
-                    updateOperation.AppendAddOp($"/{componentName}/{targetTemperaturePropertyName}", desiredTargetTemperature);
+                    updateOperation.AppendAddPropertyOp($"/{Thermostat1Component}/{targetTemperaturePropertyName}", desiredTargetTemperature);
                 }
             }
             else
             {
-                // Thermostat1 is not present in the TemperatureController twin. We will add the component.
-                var componentProperty = new Dictionary<string, object> { { targetTemperaturePropertyName, desiredTargetTemperature } };
-                var componentValuePatch = PnpHelper.CreatePatchValueForComponentUpdate(componentProperty);
-                _logger.LogDebug($"The component {componentName} does not exist on the {_digitalTwinId} digital twin.");
+                // Thermostat1 is not present in the TemperatureController twin. We will add the component
+                var componentProperty = new Dictionary<string, object> { { targetTemperaturePropertyName, desiredTargetTemperature }, { "$metadata", new object() } };
+                _logger.LogDebug($"The component {Thermostat1Component} does not exist on the {_digitalTwinId} digital twin.");
 
                 // The property path to be replaced should be prepended with a '/'
-                updateOperation.AppendAddOp($"/{componentName}", componentValuePatch);
+                updateOperation.AppendAddComponentOp($"/{Thermostat1Component}", componentProperty);
             }
 
-            _logger.LogDebug($"Update the {targetTemperaturePropertyName} property under component {componentName} on the {_digitalTwinId} digital twin to {desiredTargetTemperature}.");
-            HttpOperationHeaderResponse<DigitalTwinUpdateHeaders> updateDigitalTwinResponse = await _digitalTwinClient.UpdateDigitalTwinAsync(_digitalTwinId, updateOperation.Serialize());
+            _logger.LogDebug($"Update the {targetTemperaturePropertyName} property under component {Thermostat1Component} on the {_digitalTwinId} `" +
+                $"digital twin to {desiredTargetTemperature}.");
+            HttpOperationHeaderResponse<DigitalTwinUpdateHeaders> updateDigitalTwinResponse = await _digitalTwinClient
+                .UpdateDigitalTwinAsync(_digitalTwinId, updateOperation.Serialize());
 
             _logger.LogDebug($"Update {_digitalTwinId} digital twin response: {updateDigitalTwinResponse.Response.StatusCode}.");
 
+            // Amount of seconds to wait after updating targetTemperature property in order to allow the device to simulate temperature change
+            const int delay = 15;
```

Comment:
we do need this in a sample? I'd rather this be small, like 2-5secs or something. Else the sample will simply sit around for nothing.

Reply:
an unresponsive sample would not be the best user experience